kotest-assertions/kotest-assertions-shared/src/commonMain/kotlin/io/kotest/matchers/should.kt

```diff
@@ -11,28 +13,11 @@ infix fun <T, U : T> T.shouldBe(expected: U?) {
       else -> {
          val actual = this
          assertionCounter.inc()
-         // if we have null and non null, usually that's a failure, but people can override equals to allow it
-         if (actual == null && expected != null && actual != expected) {
-            errorCollector.collectOrThrow(actualIsNull(expected))
-         } else if (actual != null && expected == null && actual != expected) {
-            errorCollector.collectOrThrow(expectedIsNull(actual))
-         } else if (actual != null && expected != null) {
-            // for two non-null values, we use the [eq] typeclass.
-            val t = eq(actual, expected)
-            if (t != null)
-               errorCollector.collectOrThrow(t)
-         }
+         eq(actual, expected)?.let { errorCollector.collectOrThrow(it) }
```

Comment:
Minor: this could be 
`eq(actual, expected)?.let(errorCollector::collectOrThrow)`

Reply:
Done